### ISSUE TITLE
Mask图

### ISSUE THREAD

**Issue by jackymoo** [Ylexx/HBPASM#1]
请问bird_mask.jpg是怎么生成的?

**Comment by Ylexx** [comment]
将mask乘到原图上，然后重新保存成图片

**Comment by jackymoo** [comment]
哦哦，好的。还有一个问题，在train_firststep这一步:我在训练过程中对每个epoch都在测试集上做了一次测试。我发现，刚开始的几个epoch就可以达到较高的准确率了，与只用HBP比高了好多，
epoch1:51.07%
epoch2:62.05%
epoch3:69.14%
epoch4:73.94%
epoch5:73.75%
........................(不会传图)
请问你跑出来的结果也是这样吗?

**Comment by Ylexx** [comment]
是收敛快一些，所以你也可以适当减少train_firststep.py中预训练的轮数。

**Comment by jackymoo** [comment]
恩恩，好的!

**Comment by jackymoo** [comment]
方便分享一下mask可视化内部分的代码吗?我还是不太会弄。

**Comment by Ylexx** [comment]
将mask与原图乘起来后通过这个函数保存下来： cv2.imwrite('birdmask.jpg', img)

**Comment by jackymoo** [comment]
mask是(B,1,14,14)维张量，怎么与原图相乘呢?

**Comment by Ylexx** [comment]
每张图（3，448, 448)， 每个mask（1,14,14）；
将mask放大到原图大小，
import torch.nn.functional as F
F.interpolate(maskcombine, scale_factor=32, mode='bilinear', align_corners=True)；

**Comment by jackymoo** [comment]
img = cv2.imread('Great_Crested_Flycatcher_0033_29959.jpg')
model = HBPASM_model.Net('resnet50-19c8e357.pth')
model.cuda()
pre_dic = torch.load('finetune.pth')
model.load_state_dict(pre_dic)
output,mask = model(data)
mask = F.interpolate(mask, scale_factor=32, mode='bilinear', align_corners=True)
birdmask = mask * img
cv2.imwrite('birdmask.jpg', birdmask)
我这样试了一下，但是没有效果。是我理解错了吗?mask和原图应该在哪个地方乘在一起呢?

**Comment by Ylexx** [comment]
你邮箱留一下吧，我给你发个代码

**Comment by jackymoo** [comment]
357639913@qq.com,谢谢啦

**Comment by yujialeimustudent** [comment]
@Ylexx ,我只能实现热图，mask图实现好久都没有成功，希望您告知。

**Comment by Ylexx** [comment]
代码附件，大致参考一下吧


------------------ Original ------------------
